frontend/public/components/utils/k8s-watch-hook.ts

```diff
@@ -212,7 +212,7 @@ export const useK8sWatchResources = <R extends ResourcesObject>(
         } else {
           acc[key] = {
             data: resources[key].isList ? [] : {},
-            loaded: true,
+            loaded: modelsLoaded,
```

Comment:
I think that everytime we get here we can say that either all models are loaded or all requested models are loaded - https://github.com/openshift/console/blob/6fc7a2df6d190b1274196166f34d484748c9bfa7/frontend/public/components/utils/k8s-watch-hook.ts#L144 . That means its safe to say resource is not being loaded anymore and the model is missing. Or I missed something, do you have a case which I can try that causes the issue ?

`modelsLoaded` is also missing in dependency array https://github.com/openshift/console/blob/6fc7a2df6d190b1274196166f34d484748c9bfa7/frontend/public/components/utils/k8s-watch-hook.ts#L230

Reply:
@jeff-phillips-18 I guess I see the problem. Can you change `hasAllModelsLoaded` to

```
const hasAllModelsLoaded = React.useMemo(
    () => modelsLoaded || Object.values(resources).every((r) => k8sModels.includes(r.kind)),
    [k8sModels, modelsLoaded, resources],
  );
```

and revert other changes ? That might help

Reply:
@jeff-phillips-18 I've updated my PR https://github.com/openshift/console/pull/6464 with some more changes which should resolve your issues. Can you try ?

Reply:
@rawagner I'll give that a try. I definitely saw instances where all models returned `loaded: true` here while `modelsLoaded` was false.

Reply:
@jeff-phillips-18 makes sense, `hasAllModelsLoaded` was broken because second part (after or operator) was just wrong.

Reply:
Yes, that fixes the issue I was seeing. Thanks, I will remove my changes to `k8s-watch-hook.ts`